src/conv.rs

```diff
@@ -436,8 +442,14 @@ pub fn map_image_copy_buffer(
 pub fn map_texture_data_layout(native: &native::WGPUTextureDataLayout) -> wgt::ImageDataLayout {
     wgt::ImageDataLayout {
         offset: native.offset,
-        bytes_per_row: NonZeroU32::new(native.bytesPerRow),
-        rows_per_image: NonZeroU32::new(native.rowsPerImage),
+        bytes_per_row: match native.bytesPerRow {
+            native::WGPU_COPY_STRIDE_UNDEFINED => None,
+            _ => NonZeroU32::new(native.bytesPerRow),
+        },
+        rows_per_image: match native.rowsPerImage {
+            native::WGPU_COPY_STRIDE_UNDEFINED => None,
+            _ => NonZeroU32::new(native.rowsPerImage),
```

Comment:
This feels like the wrong behavior for size = 0?

Reply:
(sorry for delay, I somehow missed this)

Wouldn't it be better to support both behaviors, it preserves backwards compatibility.
It requires `NonZero` types anyway or would you prefer unwrap-ing here and only support
new behavior.

Reply:
I would solely go based on what webgpu-headers expects - my hunch is that 0 for whole length isn't what it wants.

Reply:
updated to only support new behavior